File: src/gridbase.Web/src/common/data/ProjectOverview.ts
export interface OverviewTableItem {
    id: number;
    name: string;
    rowCount: number;
    columnCount: number;
} 
export interface ProjectOverview {
    projectId: number;
    projectName: string;
    plan: string;
    tableCount: number;
    totalRows: number;
    fileCount: number;
    storageBytes: number;
    activeKeyCount: number;
    maxTables: number;
    maxStorageMb: number;
    recentTables: OverviewTableItem[];
    createdAt: string;
}

File: src/gridbase.DTO/DTOs/ProjectOverview.cs
namespace gridbase.DTO.DTOs;

public sealed class ProjectOverviewResponse
{
    public long ProjectId { get; set; }
    public string ProjectName { get; set; } = null!;
    public string Plan { get; set; } = null!;

    public int TableCount { get; set; }
    public int TotalRows { get; set; }
    public int FileCount { get; set; }
    public long StorageBytes { get; set; }
    public int ActiveKeyCount { get; set; }

    public int MaxTables { get; set; }
    public int MaxStorageMb { get; set; }

    public List<OverviewTableItem> RecentTables { get; set; } = new();

    public DateTime CreatedAt { get; set; }
}

public sealed class OverviewTableItem
{
    public long Id { get; set; }
    public string Name { get; set; } = null!;
    public int RowCount { get; set; }
    public int ColumnCount { get; set; }
}